vortex-scan/src/range_scan.rs

```diff
@@ -58,20 +58,21 @@ pub enum NextOp {
 // `evaluate(row_mask, expr)` API.
 impl RangeScanner {
     pub(crate) fn new(scan: Arc<Scanner>, row_offset: u64, mask: FilterMask) -> Self {
-        let state = scan
-            .filter()
-            .map(|filter| {
-                // If we have a filter expression, then for now we evaluate it against all rows
-                // of the range.
-                // TODO(ngates): we should decide based on mask.true_count() whether to include the
-                //  current mask or not. But note that the resulting expression would need to be
-                //  aligned and intersected with the given mask.
-                State::FilterEval((FilterMask::new_true(mask.len()), filter.clone()))
-            })
-            .unwrap_or_else(|| {
-                // If there is no filter expression, then we immediately perform a mask + project.
-                State::Project((mask.clone(), scan.projection().clone()))
-            });
+        let state = if let Some(conjunct) = scan.filter.first() {
+            // If we have a filter expression, then for now we evaluate it against all rows
+            // of the range.
+            // TODO(ngates): we should decide based on mask.true_count() whether to include the
+            //  current mask or not. But note that the resulting expression would need to be
+            //  aligned and intersected with the given mask.
+            State::FilterEval((
+                FilterMask::new_true(mask.len()),
+                conjunct.clone(),
+                scan.filter.iter().skip(1).cloned().collect(),
```

Comment:
Actually you may want to reverse this so you can pop from the end more cheaply than cloning everything to skip the first one.